src/ffi/logging.rs

```diff
@@ -164,6 +187,48 @@ fn filter_log_message(msg: String) -> Option<String> {
     None
 }
 
+enum LogMessage {
+    Message(TelioLogLevel, String),
+    Quit,
+}
+
+fn start_async_logger(cb: Box<dyn TelioLoggerCb>, buffer_size: usize) -> SyncSender<LogMessage> {
+    let (sender, receiver) = sync_channel::<LogMessage>(buffer_size);
+    let handle = Builder::new()
+        .name("libtelio-logger".to_owned())
+        .spawn(move || {
```

Comment:
Moving logging into a thread is a nice move, especially since we don't control the callback :+1: 

Derivative question if you can answer it - is this the same for event callbacks, are those in a separate thread?